Makefile

```diff
@@ -144,8 +143,50 @@ lint: lint-go lint-js
 dump-test-schema:
 	go run ./tools/dbutils ./server/datastore/mysql/schema.sql
 
-test-go: dump-test-schema generate-mock
-	go test -tags full,fts5,netgo ${GO_TEST_EXTRA_FLAGS_VAR} -parallel 8 -coverprofile=coverage.txt -covermode=atomic -coverpkg=github.com/fleetdm/fleet/v4/... ./cmd/... ./ee/... ./orbit/pkg/... ./orbit/cmd/orbit ./pkg/... ./server/... ./tools/...
+
+# This is the base command to run Go tests.
+# Wrap this to run tests with presets (see `run-go-tests` and `test-go` targets).
+# PKG_TO_TEST: Go packages to test, e.g. "server/datastore/mysql".  Separate multiple packages with spaces.
+# TESTS_TO_RUN: Name specific tests to run in the specified packages.  Leave blank to run all tests in the specified packages.
+# GO_TEST_EXTRA_FLAGS: Used to specify other arguments to `go test`.
+# GO_TEST_MAKE_FLAGS: Internal var used by other targets to add arguments to `go test`.
+#						 
+PKG_TO_TEST := "" # default to empty string; can be overridden on command line.
+go_test_pkg_to_test := $(addprefix ./,$(PKG_TO_TEST)) # set paths for packages to test
+dlv_test_pkg_to_test := $(addprefix github.com/fleetdm/fleet/v4/,$(PKG_TO_TEST)) # set URIs for packages to debug
+
+.run-go-tests:
+ifeq ($(PKG_TO_TEST), "")
+		@echo "Please specify one or more packages to test with argument PKG_TO_TEST=\"/path/to/pkg/1 /path/to/pkg/2\"..."; 
+else
+		@echo Running Go tests with command:
+		go test -tags full,fts5,netgo -run=${TESTS_TO_RUN} ${GO_TEST_MAKE_FLAGS} ${GO_TEST_EXTRA_FLAGS} -parallel 8 -coverprofile=coverage.txt -covermode=atomic -coverpkg=github.com/fleetdm/fleet/v4/... $(go_test_pkg_to_test)
+endif
+
+# This is the base command to debug Go tests.
+# Wrap this to run tests with presets (see `debug-go-tests`)
+# PKG_TO_TEST: Go packages to test, e.g. "server/datastore/mysql".  Separate multiple packages with spaces.
+# TESTS_TO_RUN: Name specific tests to debug in the specified packages.  Leave blank to debug all tests in the specified packages.
+# DEBUG_TEST_EXTRA_FLAGS: Internal var used by other targets to add arguments to `dlv test`.
+# GO_TEST_EXTRA_FLAGS: Used to specify other arguments to `go test`.
+.debug-go-tests:
+ifeq ($(PKG_TO_TEST), "")
+		@echo "Please specify one or more packages to debug with argument PKG_TO_TEST=\"/path/to/pkg/1 /path/to/pkg/2\"..."; 
+else
+		@echo Debugging tests with command:
+		dlv test ${dlv_test_pkg_to_test} --api-version=2 --listen=127.0.0.1:61179 ${DEBUG_TEST_EXTRA_FLAGS} -- -test.v -test.run=${TESTS_TO_RUN} ${GO_TEST_EXTRA_FLAGS} 
```

Comment:
I like using the command line for `dlv`, but I know some folks use vscode, do we want to add a launch.json entry to connect to this?
```
        {
            "name": "Launch Remote",
            "type": "go",
            "request": "attach",
            "mode": "remote",
            "remotePath": "${workspaceFolder}",
            "port": 61179,
            "host": "localhost",
            "cwd": "${workspaceFolder}",
            "trace": "verbose"
         }
```

Reply:
"Attach to a running Fleet server" is in the current launch.json.  It doesn't have verbose logging on, but I'm not sure I'd want to unilaterally add that in case anyone's used to the way it works now.  Although in my experience debugging go in VSCode is extremely slow so there might not be too many people using it.

Reply:
oh I totally missed that entry 🤦 . I think "trace": "verbose" isn't necessary.